File: budget/models/google.py
from typing import NamedTuple, Self

GoogleSheetRow = list[str | float | int]


class Category(NamedTuple):
    category: str | None
    name: str | None

    @classmethod
    def from_row(cls, row: list[str]) -> Self:
        checked_row = [*row[1:], None, None]
        return cls(category=checked_row[0], name=checked_row[1])

File: internal/models/google.go
// Package models provides data structures for the budget importer.
package models

// Category represents a category mapping from the lookup sheet.
type Category struct {
	Category *string // Category name.
	Name     *string // Display name override for payee.
}

// GoogleSheetRow represents a row to be inserted into Google Sheets.
type GoogleSheetRow []any
